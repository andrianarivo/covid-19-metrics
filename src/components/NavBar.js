import { NavLink } from 'react-router-dom';

export default function NavBar() {
  const links = [
    { path: '/', name: 'Home', testId: 'home-link' },
  ];
  return (
    <nav>
      <ul>
        {links.map((link) => (
          <li key={link.path}>
            <NavLink data-testid={link.testId} to={link.path}>{link.name}</NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
}
