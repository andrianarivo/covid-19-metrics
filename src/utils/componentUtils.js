import asia from '../images/asia.png';
import africa from '../images/africa.png';
import northAmerica from '../images/north_america.png';
import southAmerica from '../images/south_america.png';
import oceania from '../images/oceania.png';
import europe from '../images/europe.png';

const getImage = (continent) => {
  switch (continent.toLowerCase()) {
    case 'asia':
      return asia;
    case 'africa':
      return africa;
    case 'north america':
      return northAmerica;
    case 'south america':
      return southAmerica;
    case 'australia-oceania':
      return oceania;
    case 'europe':
      return europe;
    default:
      return asia;
  }
};

export default getImage;
