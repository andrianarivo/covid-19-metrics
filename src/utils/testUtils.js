import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';

export function withRouter(children) {
  return (
    <BrowserRouter>
      {children}
    </BrowserRouter>
  );
}

export function withProvider(children, mockStore = store) {
  return (
    <Provider store={mockStore}>
      {children}
    </Provider>
  );
}
