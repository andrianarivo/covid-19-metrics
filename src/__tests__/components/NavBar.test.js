import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import NavBar from '../../components/NavBar';
import { withRouter } from '../../utils/testUtils';

describe('Testing NavBar component', () => {
  test('renders correctly', () => {
    const tree = renderer.create(withRouter(<NavBar />));
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('user interactions', async () => {
    render(withRouter(<NavBar />));
    const homeLink = screen.getByTestId('home-link');
    await userEvent.click(homeLink);
    expect(homeLink).toBeEnabled();
  });
});
