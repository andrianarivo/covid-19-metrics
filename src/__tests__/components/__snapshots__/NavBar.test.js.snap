// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NavBar component renders correctly 1`] = `
<nav>
  <ul>
    <li>
      <a
        aria-current="page"
        className="active"
        data-testid="home-link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
  </ul>
</nav>
`;
